server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -141,6 +154,25 @@ public static void download(TranslogTransferManager translogTransferManager, Pat
         }
     }
 
+    private static void deleteTranslogFilesNotUploaded(Path location, long uploadedGeneration) throws IOException {
```

Comment:
If I recall the earlier check on deleting files before downloading translog files was added as otherwise the download used to fail, right? Now we are deleting files that are greater than the max generation referenced by the remote translog metadata file. While this is really a good optimisation, I think we should check the local file's checksum and the expected checksum. If there is a story on the checksum part already, then do let me know. If not, let's follow it up in next PR.

Reply:
On high level, comparing checksum makes sense. This is what we do for segments and can be added here as well.
The only reason I have not added this check was due to the invariant that we have: only one node will be uploading a translog file at any given time. So, if the translog file name matches, checksum must be same. Let me create a tracking issue to discuss this in more detail.

Reply:
https://github.com/opensearch-project/OpenSearch/issues/6896